src/hooks/__internal__/useMenuKeyboardNavigation/useMenuKeyboardNavigation.test.tsx

```diff
@@ -31,6 +31,8 @@ const MockComponent = ({ hideCb }: MockComponentProps) => {
 
   const getButtonChildren = useCallback(
     () =>
+      // DOM query is not used by a test to target elements
```

Comment:
Although not ideal, this query selector is used within a tested component instead of the test code itself.

Reply:
question: it took a few reads of this comment to understand what was meant

Could we be a little more verbose with the comment so it's clearer what is happening?

Reply:
I've made some adjustments to the comment. It might be a good idea for us to reconsider the `useMenuKeyboardNavigation` hook in a broader context. I feel the functionality it offers would be more effectively integrated into a single, versatile modal component or hook that encompasses all menu functionalities.